incubator/hnc/internal/validators/anchor.go

```diff
@@ -79,8 +79,13 @@ func (v *Anchor) handle(req *anchorRequest) admission.Response {
 	switch req.op {
 	case k8sadm.Create:
 		// Can't create subnamespaces in excluded namespaces
-		if config.EX[pnm] {
-			msg := fmt.Sprintf("The namespace %s is not allowed to create subnamespaces. Please create subnamespaces in a different namespace.", pnm)
+		if config.ExcludedNamespaces[pnm] {
+			msg := fmt.Sprintf("The namespace %q is not allowed to create subnamespaces. Please create subnamespaces in a different namespace.", pnm)
+			return deny(metav1.StatusReasonForbidden, msg)
+		}
+		// Can't create subnamespaces using excluded namespace names
+		if config.ExcludedNamespaces[cnm] {
```

Comment:
Do we not have a check that you can't create an anchor for a namespace that already exists (unless that namespace is a subns that's lost its anchor)?

Reply:
We do, but we use forest `cns.Exists()` right below this block. The `setExist()` is in HC reconciler but only when the namespace is not excluded (there's an early exit at the top of `Reconcile()` for excluded namespaces).

Reply:
Maybe worth a comment?